datadogclient/datadog_client_test.go

```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
-		})
-
-		err := c.PostMetrics()
-		Expect(err).ToNot(HaveOccurred())
-
-		Eventually(bodies).Should(HaveLen(1))
-
-		var payload datadogclient.Payload
-		err = json.Unmarshal(bodies[0], &payload)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(5))
-		counterNameFound := false
-		for _, metric := range payload.Series {
-			Expect(metric.Type).To(Equal("gauge"))
-
-			if metric.Metric == "datadog.nozzle.origin.counterName" {
-				counterNameFound = true
+			} else if metric.Tags[0] == "test_tag:2" {
+				tag2Found = true
 				Expect(metric.Points).To(Equal([]metrics.Point{
 					metrics.Point{
-						Timestamp: 1,
-						Value:     5.0,
+						Timestamp: 1000,
+						Value:     0.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     1.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     2.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     3.0,
 					},
 					metrics.Point{
-						Timestamp: 2,
-						Value:     11.0,
+						Timestamp: 1000,
+						Value:     4.0,
 					},
 				}))
 			}
 		}
-		Expect(counterNameFound).To(BeTrue())
-		validateMetrics(payload, 2, 0)
 
-		err = c.PostMetrics()
-		Expect(err).ToNot(HaveOccurred())
-
-		Eventually(bodies).Should(HaveLen(2))
-
-		err = json.Unmarshal(bodies[1], &payload)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(3))
-
-		validateMetrics(payload, 2, 5)
+		Expect(tag1Found).To(BeTrue())
+		Expect(tag2Found).To(BeTrue())
 	})
 
-	It("sends a value 1 for the slowConsumerAlert metric when consumer error is set", func() {
-		c.AlertSlowConsumerError()
+	It("posts ValueMetrics in JSON format & adds the metric prefix", func() {
+		k, v := makeFakeMetric("valueName", 1, 5, events.Envelope_ValueMetric, defaultTags)
+		metricsMap.Add(k, v)
+		k, v = makeFakeMetric("valueName", 2, 76, events.Envelope_ValueMetric, defaultTags)
+		metricsMap.Add(k, v)
 
-		err := c.PostMetrics()
+		err := c.PostMetrics(metricsMap)
 		Expect(err).ToNot(HaveOccurred())
 
 		Eventually(bodies).Should(HaveLen(1))
+
 		var payload datadogclient.Payload
 		err = json.Unmarshal(bodies[0], &payload)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(3))
+		Expect(payload.Series).To(HaveLen(1))
 
-		errMetric := findSlowConsumerMetric(payload)
-		Expect(errMetric).NotTo(BeNil())
-		Expect(errMetric.Type).To(Equal("gauge"))
-		Expect(errMetric.Points).To(HaveLen(1))
-		Expect(errMetric.Points[0].Value).To(BeEquivalentTo(1))
+		metric := payload.Series[0]
+		Expect(metric.Type).To(Equal("gauge"))
+		Expect(metric.Metric).To(Equal("datadog.nozzle.valueName"))
+		Expect(metric.Points).To(Equal([]metrics.Point{
+			metrics.Point{
+				Timestamp: 1,
+				Value:     5.0,
+			},
+			metrics.Point{
+				Timestamp: 2,
+				Value:     76.0,
+			},
+		}))
+		Expect(metric.Tags).To(Equal(defaultTags))
 	})
 
-	It("sends a value 0 for the slowConsumerAlert metric when consumer error is not set", func() {
```

Comment:
ditto on this I think this is an important test, this is the internal nozzle metric that says that the nozzle is misbehaving or behaving. It should be retained if possible

Reply:
Similar to above, this is now covered by https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L198

But again, could be separated into it's own test.

Reply:
That's fine!